lua/luasnip/util/util.lua

```diff
@@ -210,16 +210,13 @@ local SELECT_RAW = "LUASNIP_SELECT_RAW"
 local SELECT_DEDENT = "LUASNIP_SELECT_DEDENT"
 local TM_SELECT = "LUASNIP_TM_SELECT"
 
-local function get_selection()
-	local ok, val = pcall(vim.api.nvim_buf_get_var, 0, SELECT_RAW)
+local function get_selection(name)
+	local ok, val = pcall(vim.api.nvim_buf_get_var, 0, name)
 	-- if one is set, all are set.
 	if ok then
-		return val,
-			vim.api.nvim_buf_get_var(0, SELECT_DEDENT),
-			vim.api.nvim_buf_get_var(0, TM_SELECT)
+		return val
 	end
-	-- not ok.
-	return {}, {}, {}
+	return {}
```

Comment:
This should (it doesn't on `master`, but this is an okay place to fix it) reset the *SELECT*-vars so the selection is only available to the current snippet. That would also require making the SELECT-vars eager, it's improbable that somebody might use two types of selection, but it should work as documented.

Reply:
Do you mean, to clean up the buf-vars once they are fetched?

Reply:
Yes, exactly